### ISSUE TITLE
Any fixes with SVG elements yet?

### ISSUE THREAD

**Issue by danboyle8637** [w3c/IntersectionObserver#376]
I saw an old issue in 2017 with Intersection Observer not working with SVG. I am still finding this to be the case. 

When I setup my observer on an <path> element of an SVG it immediately kicks off an entry even when the SVG isn't on screen yet. 

And it never records any further entries or isIntersecting.

I am currently testing in Chrome.

**Comment by danboyle8637** [comment]
Just messing around and so far I have found the observer will observe the <svg> element as a whole, but not any of it's inner elements.

**Comment by 1sd15** [comment]
Any further updates on this? 
Still got the issue myself. 
Need it for Lazy Loading the Image Elements in the SVG as they are CMS/Client Side and often rather large and it speeds up the site a lot to defer them.

**Comment by danboyle8637** [comment]
@1sd15 I do have an implementation that works. But I have to look back to see if I can set it on the SVG. 

Most of the time I am observing a wrapping section or div. 

I will check it out and get back to you. Sorry for the late response.

**Comment by prat391** [comment]
Hi,

I'm facing the exact same issue. I'm trying to use Intersection Observer to delay animations on multiple elements inside an SVG. While IntersectionObserver is firing on the top level svg, it doesn't on the child level elements.

Any help would be appreciated, thanks.

**Comment by danboyle8637** [comment]
What I have been doing is using another element to listen to with my observer. Usually I will listen to the main SVG element itself and then orchestrate the animations from there. Lately I have been using Framer Motion in my React apps and sites and it makes orchestrating these animations very easy.

**Comment by prat391** [comment]
hi! 

I am an absolute beginner, so even though in theory i understand what you're saying, i won't be able to implement it :/

My question is, can i use either just the parent svg class to trigger the entire animation, or use a separate element altogether to trigger the animation?

What would the code look like for this? I am a beginner so i am having a tough time figuring this out. In layman terms, i want this: when x comes into view, trigger all the animations for Y.

Thanks

**Comment by Sonny1314** [comment]
@danboyle8637 I have the same issue. I have pasted SVG file itself in HTML markup. Instead, I tried to put my SVG inside a section tag and use intersection observer to observe the section tag. I tried to toggle animation class names inside the SVG elements when the intersecting ratio becomes more than one. But somehow it is not working.
The FOREGROUND is class in my SVG file. and section is the id of the section in which I have pasted the code of SVG file

JS code
![image](https://user-images.githubusercontent.com/50945527/85792018-68b23e80-b750-11ea-907c-7c6f12282a5d.png)

HTML code
![image](https://user-images.githubusercontent.com/50945527/85792096-8e3f4800-b750-11ea-871b-d52ef0a79234.png)

in getting good output in my intersection observer in the console but it's not working as it should. 
![image](https://user-images.githubusercontent.com/50945527/85792432-1b829c80-b751-11ea-9f7f-4e46df53c88e.png)

**Comment by danboyle8637** [comment]
Sorry I am working on a project with a deadline of this weekend. I will show you how I built my intersection observer in Hugo. I think that is the closest to vanilla JS and HTML. I will make a video and also post some code on a CodeSandbox. Sorry but I am under a serious deadline.

**Comment by Sonny1314** [comment]
@prat391 if you are working with basic HTML and javascript then I can send you a code snippet of how I achieved it for my project. I am very new to React so I won't be of much help. Drop a message if you want code snippet. That might help you.

**Comment by prat391** [comment]
@Sonny1314 would greatly appreciate it thanks!

**Comment by Sonny1314** [comment]
I have also provided codePen link below. Hope it helps.
So here is the basic layout of my code structure. I have pasted an SVG file inside my HTML document directly. You can use <object> element to insert as well but the js code will change slightly.  So I have Two sections _and I have nested the SVG inside the second section of my code. I have used Intersection observer to observe the **Second section**
 ie. `<section id="section2">**MY SVG CODE HERE**<section/> `
and once the intersection ratio of the second section is above 0 I have used javascript to toggle classes (animation class defined in my CSS)   and the SVG elements.

below are the code snippets.
**HTML CODE**
![image](https://user-images.githubusercontent.com/50945527/85850548-a358bd00-b7ca-11ea-9e41-b6ce811a87c4.png)
..inline css for filling svg elements and then comes picture of the svg elemets
![image](https://user-images.githubusercontent.com/50945527/85850634-cedba780-b7ca-11ea-9f05-b8421835c7cf.png)

`
MY CSS CLASSES FOR ANIMATION`
![image](https://user-images.githubusercontent.com/50945527/85850694-efa3fd00-b7ca-11ea-82ef-42296a1bc290.png)



**JS CODE:**
![image](https://user-images.githubusercontent.com/50945527/85852522-67275b80-b7ce-11ea-989e-1089c043db9e.png)

I have used `window.addEventListener("load", function() { ` so that everything is executed after the whole page loads.
Then coming to line 7 I have stored value of my section in which SVG is located in a variable called section.
I created a new intersection observer and Iam observing this section variable as you can see in line 30.

`var TreeLeft = document.getElementById("_3TreeLeft");
// left tree is element in my svg 
if(entries[0].intersectionRatio > 0){
        
             TreeLeft.classList.toggle("slide-in-bck-bl-left-tree"); 
             house.classList.toggle("slide-in-bck-bl-house");
             TreeeRight.classList.toggle("slide-in-bck-bl-right-tree");
             moon.classList.toggle("Moon");
           
         }`
Now when the intersection ratio is more than one I select the element in SVG and toggle an animation class to it.

The CodePen link is below, make sure to switch to the sideView
 in CodePen to make viewport height big. 

**CODEPEN LINK:** https://codepen.io/sonny1314/pen/GRovpEQ

![image](https://user-images.githubusercontent.com/50945527/85852237-cfc20880-b7cd-11ea-9903-1349629dafcf.png)

**Comment by szager-chromium** [comment]
Just to confirm the discussion thus far: chromium does not currently support observing an svg element other thant the top-level <svg>. There is a chromium bug tracking this:

https://bugs.chromium.org/p/chromium/issues/detail?id=963246

As this is not an issue with the spec text, I am closing this issue.
